kv: TestFlowControlGranterAdmitOneByOne failed

This test fails about once a day with the same error.

https://teamcity.cockroachdb.com/test/1304895367360427992?currentProjectId=Cockroach_Ci_TestsGcpLinuxX8664BigVm&expandTestHistoryChartSection=true&orderBy=status&order=desc

```
Failed
=== RUN   TestFlowControlGranterAdmitOneByOne
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/efdc05cf79b5dfd2ad5dc565351e4382/logTestFlowControlGranterAdmitOneByOne3800129376
    test_log_scope.go:81: use -show-logs to present logs inline
    datadriven.go:144:
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/13216/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/testdata/flow_control_integration/granter_admit_one_by_one:1:
        output didn't match expected:
        @@ -9,27 +9,27 @@
            WHERE name LIKE '%kvadmission%tokens%'
         ORDER BY name ASC;
           kvadmission.flow_controller.elastic_tokens_available   | 21 MiB
           kvadmission.flow_controller.elastic_tokens_deducted    | 3.0 MiB
        -  kvadmission.flow_controller.elastic_tokens_returned    | 0 B
        +  kvadmission.flow_controller.elastic_tokens_returned    | 3.0 KiB
           kvadmission.flow_controller.elastic_tokens_unaccounted | 0 B
           kvadmission.flow_controller.regular_tokens_available   | 45 MiB
           kvadmission.flow_controller.regular_tokens_deducted    | 3.0 MiB
        -  kvadmission.flow_controller.regular_tokens_returned    | 0 B
        +  kvadmission.flow_controller.regular_tokens_returned    | 3.0 KiB
           kvadmission.flow_controller.regular_tokens_unaccounted | 0 B
         -- Observe the total tracked tokens per-stream on n1.
         SELECT range_id, store_id, crdb_internal.humanize_bytes(total_tracked_tokens::INT8)
             FROM crdb_internal.kv_flow_control_handles
           range_id | store_id | total_tracked_tokens
         -----------+----------+-----------------------
        -  65       | 1        | 1.0 MiB
        -  65       | 2        | 1.0 MiB
        -  65       | 3        | 1.0 MiB
        +  65       | 1        | 1023 KiB
        +  65       | 2        | 1023 KiB
        +  65       | 3        | 1023 KiB
         -- (Allow below-raft admission to proceed.)
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/efdc05cf79b5dfd2ad5dc565351e4382/logTestFlowControlGranterAdmitOneByOne3800129376
--- FAIL: TestFlowControlGranterAdmitOneByOne (3.92s)
```

Jira issue: CRDB-31423

repro on m1 did not work but it failed within 1k runs on gce worker:
```
FAIL: //pkg/kv/kvserver:kvserver_test (run 677 of 1000) (see /home/aadityasondhi/.cache/bazel/_bazel_aadityasondhi/ef545d1c1fadb47793aee4447a30c68c/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/testlogs/pkg/kv/kvserver/kvserver_test/run_677_of_1000/test.log)
ERROR: /home/aadityasondhi/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/BUILD.bazel:243:8: Testing //pkg/kv/kvserver:kvserver_test (run 677 of 1000) failed: Test failed, aborting
==================== Test output for //pkg/kv/kvserver:kvserver_test (run 677 of 1000):
I230925 19:50:33.873462 1 (gostd) rand.go:250  [-] 1  random seed: 1054950456997938669
test logs left over in: /tmp/cockroach/_tmp/1eef343fabc0819a524e89405f54d392/logTestFlowControlGranterAdmitOneByOne3684591267
--- FAIL: TestFlowControlGranterAdmitOneByOne (2.89s)
    test_log_scope.go:167: test logs captured to: /tmp/cockroach/_tmp/1eef343fabc0819a524e89405f54d392/logTestFlowControlGranterAdmitOneByOne3684591267
    test_log_scope.go:81: use -show-logs to present logs inline
    datadriven.go:144:
        /home/aadityasondhi/.cache/bazel/_bazel_aadityasondhi/ef545d1c1fadb47793aee4447a30c68c/sandbox/linux-sandbox/3632/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/testdata/flow_control_integration/granter_admit_one_by_one:1:

        output didn't match expected:
        @@ -9,27 +9,27 @@
            WHERE name LIKE '%kvadmission%tokens%'
         ORDER BY name ASC;

           kvadmission.flow_controller.elastic_tokens_available   | 21 MiB
           kvadmission.flow_controller.elastic_tokens_deducted    | 3.0 MiB
        -  kvadmission.flow_controller.elastic_tokens_returned    | 0 B
        +  kvadmission.flow_controller.elastic_tokens_returned    | 9.0 KiB
           kvadmission.flow_controller.elastic_tokens_unaccounted | 0 B
           kvadmission.flow_controller.regular_tokens_available   | 45 MiB
           kvadmission.flow_controller.regular_tokens_deducted    | 3.0 MiB
        -  kvadmission.flow_controller.regular_tokens_returned    | 0 B
        +  kvadmission.flow_controller.regular_tokens_returned    | 9.0 KiB
           kvadmission.flow_controller.regular_tokens_unaccounted | 0 B


         -- Observe the total tracked tokens per-stream on n1.
         SELECT range_id, store_id, crdb_internal.humanize_bytes(total_tracked_tokens::INT8)
             FROM crdb_internal.kv_flow_control_handles

           range_id | store_id | total_tracked_tokens
         -----------+----------+-----------------------
        -  65       | 1        | 1.0 MiB
        -  65       | 2        | 1.0 MiB
        -  65       | 3        | 1.0 MiB
        +  65       | 1        | 1021 KiB
        +  65       | 2        | 1021 KiB
        +  65       | 3        | 1021 KiB


         -- (Allow below-raft admission to proceed.)


    panic.go:522: -- test log scope end --
FAIL
I230925 19:50:36.783570 1 (gostd) testmain.go:1614  [-] 1  Test //pkg/kv/kvserver:kvserver_test exited with error code 1
================================================================================
ERROR: Test failed, aborting
INFO: Elapsed time: 404.325s, Critical Path: 227.82s
INFO: 3701 processes: 78 internal, 3623 linux-sandbox.
//pkg/kv/kvserver:kvserver_test                                       NO STATUS
  /home/aadityasondhi/.cache/bazel/_bazel_aadityasondhi/ef545d1c1fadb47793aee4447a30c68c/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/testlogs/pkg/kv/kvserver/kvserver_test/run_677_of_1000/test.log

Executed 1 out of 1 test: 1 was skipped.
There were tests whose specified size is too big. Use the --test_verbose_timeout_warnings command line option to see which ones these are.
INFO: Build Event Protocol files produced successfully.
INFO: Build completed successfully, 3701 total actions
ERROR: exit status 3
aadityasondhi@gceworker-aadityasondhi:~/go/src/github.com/cockroachdb/cockroach$
```

Going to dig deeper

--- a/pkg/kv/kvserver/flow_control_integration_test.go
+++ b/pkg/kv/kvserver/flow_control_integration_test.go
@@ -2331,11 +2331,13 @@ func newFlowControlTestHelper(t *testing.T, tc *testcluster.TestCluster) *flowCo
 }
 
 func (h *flowControlTestHelper) init() {
-	if _, err := h.tc.Conns[0].Exec(`SET CLUSTER SETTING kvadmission.flow_control.enabled = true`); err != nil {
-		h.t.Fatal(err)
-	}
-	if _, err := h.tc.Conns[0].Exec(`SET CLUSTER SETTING kvadmission.flow_control.mode = 'apply_to_all'`); err != nil {
-		h.t.Fatal(err)
+	// Reach into each server's cluster setting and override. This causes any
+	// registered change callbacks to run immediately, which is important since
+	// running them with some lag (which happens when using SQL and `SET CLUSTER
+	// SETTING`) interferes with the later activities in these tests.
+	for _, s := range h.tc.Servers {
+		kvflowcontrol.Enabled.Override(context.Background(), &s.ClusterSettings().SV, true)
+		kvflowcontrol.Mode.Override(context.Background(), &s.ClusterSettings().SV, int64(kvflowcontrol.ApplyToAll))
 	}
 }
 

